meson.build

```diff
@@ -47,6 +47,24 @@ endif
 cy = meson.get_compiler('cython')
 if cy.version().version_compare('>=3.1.0')
     add_project_arguments('-Xfreethreading_compatible=true', language: 'cython')
+
+    # Use shared utility code to reduce wheel sizes
```

Comment:
Cool seems useful. Is there any discussion to add this natively to Meson instead of custom-coding this ourselves?

Reply:
I don't see how that's possible, since it generates an extra extension module and projects may have their code set up (e.g., with a Meson generator) to apply Cython/C compile args to all their extension modules. They may also have opinions about where the extension module should live in the import tree.